Add an admin API for users' media statistics (#8700)
matrix-org/synapse: docs/admin_api/statistics.md

# Users' media usage statistics

> Gets information about all local media usage of all users. Gives the

Reviewer: ```suggestion
Returns information about all local media usage of users. Gives the
```

  used for pagination, denoting the maximum number of items to return
  in this call. Defaults to `100`.
* `from`: string representing a positive integer - Is optional but used for pagination,
  denoting the offset in the returned results. This should be treated as an opaque value
  and not explicitly set to anything other than the return value of `next_token` from a
  previous call. Defaults to `0`.
* `order_by` - string - The method in which to sort the returned list of users. Valid values are:
  - `user_id` - Users are ordered alphabetically by `user_id`. This is the default.
  - `displayname` - Users are ordered alphabetically by `displayname`.
  - `media_length` - Users are ordered by the total size of uploaded media in bytes.
    Smallest to largest.
  - `media_count` - Users are ordered by number of uploaded media. Smallest to largest.
* `from_ts` - string representing a positive integer - Considers only
  files created at this timestamp or later. Unix timestamp in ms.
* `until_ts` - string representing a positive integer - Considers only
  files created at this timestamp or earlier. Unix timestamp in ms.

> * `search_term` - string - Filter users by their user ID localparts **or** `displayname`.
>   Search term can be contained in any part of the user ID localparts or `displayname`.

Reviewer: ```suggestion
* `search_term` - string - Filter users by their user ID localpart **or** displayname.
  The search term can be found in any part of the string.
```

  Defaults to no filtering.
* `dir` - string - Direction of order. Either `f` for forwards or `b` for backwards.
  Setting this value to `b` will reverse the above sort order. Defaults to `f`.

**Response**

The following fields are returned in the JSON response body:

* `users` - An array of objects, each containing information
  about the user and their local media. Objects contain the following fields:
  - `displayname` - string - Displayname of this user.
  - `media_count` - integer - Number of uploaded media by this user.
  - `media_length` - integer - Size of uploaded media in bytes by this user.

>   - `user_id` - string - Fully-qualified user id: for example, @user:server.com.

Reviewer: ```suggestion
  - `user_id` - string - Fully-qualified user ID (ex. `@user:server.com`).
```

> * `next_token` - integer - Indication for pagination. See above.

Reviewer: ```suggestion
* `next_token` - integer - Opaque value used for pagination. See above.
```

> * `total` - integer - Total number of users.

Reviewer: ```suggestion
* `total` - integer - Total number of users after filtering.
```
